learning_resources/views.py

```diff
@@ -227,6 +227,35 @@ def upcoming(self, request: Request) -> QuerySet:  # noqa: ARG002
         return self.get_paginated_response(serializer.data)
 
 
+class PopularResourcesViewSetMixin(GenericAPIView):
+    """ViewSet mixin for adding popular resource functionality."""
+
+    resource_type_name_plural: str
+
+    def __init_subclass__(cls) -> None:
+        """Initialize subclasses by updating the view with the correct serializer."""
+        name = cls.resource_type_name_plural
+        # this decorator mutates the view in place so the return value is safely ignored
+        extend_schema_view(
+            new=extend_schema(
+                description=f"Get a paginated list of popular {name}.",
+                responses=cls.serializer_class(many=True),
+            ),
+        )(cls)
+        super().__init_subclass__()
+
+    @extend_schema(summary="List Popular")
+    @action(methods=["GET"], detail=False, name="Popular Resources")
+    def popular(self, request: Request) -> QuerySet:  # noqa: ARG002
```

Comment:
This appears to also be applied to /api/v1/learning_resources_search/popular which returns odd results for the views attribute at the moment. we should either resolve it so it works with learning_resources_search or remove the /popular from the opensearch endpoint.

Reply:
This was a bit confusing because (other than the constants that got added) I didn't touch this at all. But: the URL path for this was using a `re_path` so putting anything after `learning_resources_search` worked. Changed this in https://github.com/mitodl/mit-open/pull/789/commits/e4de0aa0185d6b8ba74bc24666497a2bf249e909 to use a normal, non-regex `path` call instead, so it won't let you add path elements after the `learning_resources_search`. 

(I am assuming this is OK - the tests pass, it wasn't exposed in the openapi schema, and intrinsically it doesn't seem like this was a real necessity, but just habit to use re_path rather than path. The other route in there is still using the `re_path` but can switch that over too if it doesn't really need a regex.)

Reply:
ah ok. interesting.